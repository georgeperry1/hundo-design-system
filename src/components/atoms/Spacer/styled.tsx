import styled from 'styled-components';

import { SpacerProps } from './index';

export const SpacerElement = styled.div`
  display: ${({ inline }: SpacerProps) => (inline ? 'inline-block' : 'block')};
  height: ${({ h }: SpacerProps) => (h ? `${h}px` : 0)};
  width: ${({ w }: SpacerProps) => (w ? `${w}px` : 0)};
`;
