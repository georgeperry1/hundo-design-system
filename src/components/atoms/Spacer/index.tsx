import { SpacerElement } from './styled';

export type SpacerProps = {
  w?: number;
  h?: number;
  inline?: boolean;
};

const Spacer: React.FC<SpacerProps> = (props) => {
  return <SpacerElement {...props} />;
};

export default Spacer;
