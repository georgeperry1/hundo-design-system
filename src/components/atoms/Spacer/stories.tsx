import { Story, Meta } from '@storybook/react';

import Spacer, { SpacerProps } from '.';

export default {
  title: 'Components/Atoms/Spacer',
  component: Spacer,
} as Meta;

const Block = () => (
  <div
    style={{
      width: '100px',
      height: '100px',
      backgroundColor: '#111111',
      borderRadius: '4px',
    }}
  ></div>
);

export const W: Story<SpacerProps> = (args) => (
  <div style={{ display: 'flex', flexDirection: 'row' }}>
    <Block />
    <Spacer {...args} />
    <Block />
  </div>
);

W.args = {
  w: 10,
  h: 0,
};

export const H: Story<SpacerProps> = (args) => (
  <div style={{ display: 'flex', flexDirection: 'column', height: '400px', width: '400px' }}>
    <Block />
    <Spacer {...args} />
    <Block />
  </div>
);

H.args = {
  w: 0,
  h: 10,
};

export const Inline: Story<SpacerProps> = (args) => (
  <p>
    Hello
    <Spacer {...args} />
    Hundo!
  </p>
);

Inline.args = {
  w: 10,
  h: 0,
  inline: true,
};
