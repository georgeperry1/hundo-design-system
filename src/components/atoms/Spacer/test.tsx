import { render } from '@testing-library/react';

import Spacer from '.';

describe('Spacer', () => {
  it('should render', () => {
    const { baseElement } = render(<Spacer />);

    expect(baseElement).toMatchSnapshot();
  });

  it('should support w and h', () => {
    const { baseElement } = render(<Spacer w={10} h={10} />);

    expect(baseElement).toMatchSnapshot();
  });

  it('should work with inline mode', () => {
    const { baseElement } = render(<Spacer w={10} inline />);

    expect(baseElement).toMatchSnapshot();
  });
});
