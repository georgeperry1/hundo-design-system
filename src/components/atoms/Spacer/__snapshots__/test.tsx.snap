// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spacer should render 1`] = `
.c0 {
  display: block;
  height: 0;
  width: 0;
}

<body>
  <div>
    <div
      class="c0"
    />
  </div>
</body>
`;

exports[`Spacer should support w and h 1`] = `
.c0 {
  display: block;
  height: 10px;
  width: 10px;
}

<body>
  <div>
    <div
      class="c0"
    />
  </div>
</body>
`;

exports[`Spacer should work with inline mode 1`] = `
.c0 {
  display: inline-block;
  height: 0;
  width: 10px;
}

<body>
  <div>
    <div
      class="c0"
    />
  </div>
</body>
`;
